hack/dockerized

```diff
@@ -77,7 +119,7 @@ _rsync --delete "rsync://root@127.0.0.1:${RSYNCD_PORT}/build/cluster/examples/"
 if [ "$SYNC_VENDOR" = "true" ]; then
     _rsync --delete "rsync://root@127.0.0.1:${RSYNCD_PORT}/vendor" "${VENDOR_DIR}/"
 fi
-# Copy the build output out of the container, make sure that _out exactly matches the build result
+# Copy the build output out of the container
 if [ "$SYNC_OUT" = "true" ]; then
-    _rsync --delete "rsync://root@127.0.0.1:${RSYNCD_PORT}/out" ${OUT_DIR}
```

Comment:
The `delete` here is very important. If you remove it you will see stale artifacts in `_out`. In `_out` we only have compiled or generated things, so we need to keep it in sync with the builder docker-volume.

Reply:
I agree that something is needed to avoid having stale artifacts, but the `delete` directive is too broad a brush given the chosen file layout.  For example, if I run `make manifests`, this deletes all the binaries, and vice-versa.  Maybe a more targeted copy would be better, but I agree that maybe this is becoming too complicated.  The important goal here that I see is to get the appropriate environment variables into the containers.

Reply:
> I agree that something is needed to avoid having stale artifacts, but the delete directive is too broad a brush given the chosen file layout. For example, if I run make manifests, this deletes all the binaries, and vice-versa.

Not really. As said, the actual building and generating is happening inside the docker container and the artifacts are stored in a docker-volume: Check `docker volume ls` for a volume called `kubevirt`.

If you run `make`, the binaries are compiled and stored in the `_out` directory inside the volume. Then the `rsync --delete` at the end ensures that `_out`  in the volume is in sync with `_out` outside (there is a little bit more to also get rid of binaries not part of that branch, but don't want to make it too complicated here). 

When we next run `make manifests`, the manifests are also generated inside the container and put in the `_out` folder in the docker volume. Then `rsync --delete` makes again sure that `_out` inside the volume is in sync with `_out` outside. **Note:** This does not clean any of the binaries which were built before. Also they would not even be copied out again by `rsync`. Only stale stuff would disappear from the `_out` directory outside. That is exactly what we want.

Finally, regarding to syncing, building or generating it does not make much difference if you run

  `hack/dockerized "hack/build.sh && hack/manifests.sh `

or
 
  `make build && make manifests`

the only difference is the overhead of starting the container and a very small rsync overhead for detecting changes.

You can also have a look yourself into the volume. Run `make && make manifests`, then run `hack/dockerized.sh /bin/sh` to enter the  container. That will directly drop you inside the volume mirror location. You will find binaries and the artifacts in `_out` there.